src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
-
+function isTrianglePossible(a: number, b: number, c: number): boolean {
+  return a + b > c && a + c > b && b + c > a;
 }
 
-export class Triangle {
+function roundDownFloat(number: number, amountOfDecimals: number = 2): number {
+  return Math.floor(number * 10 ** amountOfDecimals) / 10 ** amountOfDecimals;
+}
 
+export interface Figure {
+  shape: string,
+  color: string,
+  getArea: Function,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((number) => number <= 0)) {
+      throw new Error("Sides of the triangle can't be equal or smaller than 0");
+    }
+
+    if (!isTrianglePossible(a, b, c)) {
+      throw new Error("One side can't be greater than the sum of two others");
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) * 0.5;
 
+    const area = (semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c)
+    ) ** 0.5;
```

Comment:
Better to use `Math.sqrt`. Read more [here](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/sqrt)